benchmark/parse_output.py

```diff
@@ -141,14 +142,109 @@ def parse_hgtector(input_f):
     return '\n'.join(hgts)
 
 
-def parse_output(hgt_results_fp, method):
+def parse_egid(input_f, genbank_fp):
+    """ Extract genes contained in GIs identified by EGID
+
+    Parameters
+    ----------
+    input_f: string
+        file descriptor for EGID output results (GI coordinates)
+    genbank_fp: string
+        file path to genome in GenBank format (containing gene coordinates)
+
+    Notes
+    -----
+    genbank_fp is the intermediate GenBank file generated by reformat_input.py,
+    in which multiple sequences are concantenated, instead of the original
+    GenBank file.
+
+    Returns
+    -------
+    output: string
+        gene names (protein_ids) separated by newline
+    """
+    genes = {}
+    gb = Sequence.read(genbank_fp, format='genbank')
+    for feature in gb.interval_metadata.features:
+        if feature['type_'] == 'CDS':
+            if 'protein_id' in feature:
+                protein_id = feature['protein_id'].replace('\"', '')
+                if protein_id not in genes:
+                    loc = gb.interval_metadata.features[feature]
+                    start = loc[0][0] + 1
+                    end = loc[0][1]
+                    genes[protein_id] = (start, end)
+    genes_in_gi = {}
+    for line in input_f:
+        l = line.strip().split()
+        if len(l) < 2:
+            continue
+        (start, end) = (int(l[0]), int(l[1]))
```

Comment:
I think the confusion comes because you are note actually creating a tuple, you are simply summarizing 2 lines of code into 1. Perhaps splitting is a good idea, obviously fine if not.